torch/testing/_internal/common_modules.py

```diff
@@ -956,4 +1025,57 @@ def module_inputs_torch_nn_LSTMCell(module_info, device, dtype, requires_grad, *
                module_inputs_func=module_inputs_torch_nn_LSTMCell),
     ModuleInfo(torch.nn.Sigmoid,
                module_inputs_func=module_inputs_torch_nn_Sigmoid),
+    ModuleInfo(torch.nn.RNN,
+               module_inputs_func=partial(module_inputs_torch_nn_RNN_GRU, is_rnn=True),
+               decorators=(
+                   # RuntimeError: Batching rule not implemented for aten::_cudnn_rnn_backward.
+                   # We could not generate a fallback
+                   DecorateInfo(
+                       unittest.expectedFailure, "TestModule", "test_grad",
+                       active_if=(TEST_CUDNN and not TEST_WITH_ROCM)
+                   ),
+                   # NotImplementedError: the derivative for '_cudnn_rnn_backward' is not implemented.
+                   # Double backwards is not supported for CuDNN RNNs due to limitations in the CuDNN API
+                   DecorateInfo(
+                       unittest.expectedFailure, "TestModule", "test_gradgrad",
+                       active_if=(TEST_CUDNN and not TEST_WITH_ROCM)
+                   ),
+                   # CUDNN RNN doesn't accept non-contiguous hx
+                   DecorateInfo(
+                       unittest.expectedFailure, "TestModule", "test_non_contiguous_tensors",
+                       active_if=(TEST_CUDNN and not TEST_WITH_ROCM)
+                   ),
+                   # MIOPEN RNN doesn't accept non-contiguous hx (this is dispatched to miopen only for float).
+                   DecorateInfo(
+                       unittest.expectedFailure, "TestModule", "test_non_contiguous_tensors",
+                       active_if=(TEST_CUDNN and TEST_WITH_ROCM), dtypes=(torch.float,)
+                   ),
```

Comment:
phew what a pain this must have been! sorry about all these failures - we'll have to address these at some point. Thanks for the descriptive comments for each!